docs/config.rst

```diff
@@ -0,0 +1,778 @@
+
+Verify the change by looking for a default option value when running ``cci task info <name>``.
+
+.. code-block:: yaml
+
+    $ cci task info run_tests
+    run_tests
+
+    Description: Runs all apex tests
+
+    Class: cumulusci.tasks.apex.testrunner.RunApexTests
+
+    Command Syntax
+
+        $ cci task run run_tests
+
+    Options
+        .
+        .
+        .
+      -o required_org_code_coverage_percent PERCENTAGE
+        Optional
+        Require at least X percent code coverage across the org following the test run.
+        Default: 90
+
+
+Add a Custom Task
+^^^^^^^^^^^^^^^^^
+
+To define a new task for your project, add the task name under the ``tasks`` section of your ``cumulusci.yml`` file.
+
+For example, let's create a custom task named ``deploy_reports`` that deploys a set of reports stored in your project's unpackaged metadata located in ``unpackaged/config/reports``.
+
+First, look up the Python class associated with the standard task ``deploy``. From there we see that the ``deploy`` task has a ``class_path`` value of ``cumulusci.tasks.salesforce.Deploy``.
+
+Store the task under the ``tasks`` section of the ``cumulusci.yml`` file.
+
+.. code-block:: yaml
+
+    deploy_reports:
+        description: Deploy Reports
+        class_path: cumulusci.tasks.salesforce.Deploy
+        group: projectName
+        options:
+            path: unpackaged/config/reports
+
+.. tip::
+
+    Be sure to include the value we retrieved for ``class_path``.
+    Also, consider adding a common ``group`` attribute to make it easier to see the tasks specific to your project when running ``cci task list``.
+
+Congratulations! You created a new custom task in CumulusCI.
+
+To write a custom task in Python and make it available to other users in the project, add the task under the ``tasks`` section of the ``cumulusci.yml`` file. (Let's assume that your task's class is named ``MyNewTaskClassName`` and exists in the file ``tasks/task_file.py``.)
+
+.. code-block:: yaml
+
+    tasks:
+        my_new_task:
+            description: Description of the task
+            class_path: tasks.task_file.MyNewTaskClassName
+            group: projectName
+
+
+
+Use Variables for Task Options
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To reference a specific value within the ``tasks`` section of the ``cumulusci.yml`` file, use the ``$project_config`` variable.
+
+For example, NPSP uses a variable for the project's namespace by setting a value of ``$project_config.project__package__namespace``. This variable is then referenced in the project's custom ``deploy_qa_config`` task where it's passed as the value for the ``namespace_inject`` option.
+
+.. note::
+
+    A double underscore (``__``) refers to a subsequent level in the ``cumulusci.yml`` file.
+
+.. code-block:: yaml
+
+    deploy_qa_config:
```

Comment:
Not a blocker for release, but we don't use this pattern anymore because namespace injection is inferred.